vmware_dvs/driver/dvs_mechanism_driver.py

```diff
@@ -118,27 +126,38 @@ def update_port_precommit(self, context):
 
     @port_belongs_to_vmware
     def update_port_postcommit(self, context):
-        self.dvs_notifier.update_postcommit_port_call(
-            context.current,
-            context.original,
-            context.network.network_segments[0],
-            context.host
-        )
-        # TODO(ekosareva): removed one more condition(is it really needed?):
-        #                  'dvs_port_key' in port['binding:vif_details']
-        if (context.current['binding:vif_type'] == 'unbound' and
-                context.current['status'] == n_const.PORT_STATUS_DOWN):
-            context._plugin.update_port_status(
-                context._plugin_context,
-                context.current['id'],
-                n_const.PORT_STATUS_ACTIVE)
+        if self._check_net_port(context):
+            self.dvs_notifier.update_postcommit_port_call(
+                context.current,
+                context.original,
+                context.network.network_segments[0],
+                context.host
+            )
+            # TODO(ekosareva): removed one more condition. Is it really needed?
+            #                  'dvs_port_key' in port['binding:vif_details']
+            if (context.current['binding:vif_type'] == 'unbound' and
+                    context.current['status'] == n_const.PORT_STATUS_DOWN):
+                context._plugin.update_port_status(
+                    context._plugin_context,
+                    context.current['id'],
+                    n_const.PORT_STATUS_ACTIVE)
 
     @port_belongs_to_vmware
     def delete_port_postcommit(self, context):
-        self.dvs_notifier.delete_port_call(context.current, context.original,
+        if self._check_net_port(context):
+            self.dvs_notifier.delete_port_call(context.current,
+                                           context.original,
                                            context.network.network_segments[0],
                                            context.host)
 
+    def _check_net_type(self, context):
+        n_type = context.network_segments[0]['network_type']
+        return n_type == constants.TYPE_VLAN
+
+    def _check_net_port(self, context):
+        n_type = context.network.network_segments[0]['network_type']
+        return n_type == constants.TYPE_VLAN
+
```

Comment:
This 2 functions is very similar, may be combine it in one with network arg. Smth like: 
def _check_net_type(self, network):
    return network.network_segments[0]['network_type'] == constants.TYPE_VLAN  
And pass context (for networks)  or context.network (for ports) as argument